### PR TITLE
[REVIEW] Handle RMM Python API Changes

### PR DESCRIPTION
**Pull request by kkraus14**
Depends on https://github.com/rapidsai/rmm/pull/171

### CODE REVIEW

File: python/cudf/cudf/utils/utils.py [hunk 1]
```diff
@@ -277,25 +276,25 @@ def _set_rmm_config(
...
         Enabling this option will introduce performance overhead.
     """
-    rmm.finalize()
-    rmm_config.use_managed_memory = use_managed_memory
     if use_pool_allocator:
-        rmm_config.use_pool_allocator = use_pool_allocator
         if initial_pool_size is None:
             initial_pool_size = 0  # 0 means 1/2 GPU memory
         elif initial_pool_size == 0:
             initial_pool_size = 1  # Since "0" is semantic value, use 1 byte
```
Comment (ayushdg) [inline python/cudf/cudf/utils/utils.py]: I believe some of this logic was introduced in rapidsai/rmm#171 which makes it tough to initialize pool mode with 1/2 gpu memory from cudf. `initial_pool_size=None` sets it to 0 here. rapidsai/rmm#171 infers `initial_pool_size` as 0 and sets it to 1, hence making it tough to initialize with the default 1/2 Gpu memory case

**Comment by ayushdg** [comment]
Just for clarification, since the `set_allocator` method is wrapped by `initfunc` and the private method no longer exists, the current way to reinitialize rmm with different modes (from cudf or Dask) would be through the `rmm.reintialize` api right?

**Comment by kkraus14** [comment]
> Just for clarification, since the `set_allocator` method is wrapped by `initfunc` and the private method no longer exists, the current way to reinitialize rmm with different modes (from cudf or Dask) would be through the `rmm.reintialize` api right?

Yes, if you need to change the allocation mode in the middle of a workflow then you'd use `rmm.reinitialize(...)`

**Comment by harrism** [comment]
@kkraus14 RMM #171 is merged, can we get this merged? I realized I just broke CI by merging that.

**Comment by harrism** [comment]
rerun tests

**Pull request merged**